src/vec.rs

```diff
@@ -534,6 +535,59 @@ where
     }
 }
 
+impl<ID, Element> IdentifiedVec<ID, Element>
+where
+    ID: Eq + Hash + Clone + Debug,
+    Element: Eq + Debug,
+{
+    /// Try append a new unique member to the end of the `identified_vec`, if the `identified_vec` already contains the Value or ID a Error will be returned.
+    ///
+    /// - Parameter item: The element to add to the `identified_vec`.
+    /// - Returns: Either a Ok() with a pair `(inserted, index)`, where `inserted` is a Boolean value indicating whether
+    ///   the operation added a new element, and `index` is the index of `item` in the resulting
+    ///   `identified_vec`. If the given ID already exist `Error::ElementWithSameIDFound` willl be returned and if the value pre-exists within the collection the function call returns `Error::ElementWithSameValueFound`.
+    /// - Complexity: The operation is expected to perform O(1) copy, hash, and compare operations on
+    ///   the `ID` type, if it implements high-quality hashing.
+    #[inline]
+    pub fn try_append_unique_element(&mut self, element: Element) -> Result<(bool, usize), Error> {
+        let id = self.id(&element);
+
+        if let Some(value) = self.get(&id) {
+            if value == &element {
+                return Err(Error::ElementWithSameValueFound(format!("{:?}", value)));
+            } else {
+                return Err(Error::ElementWithSameIDFound(format!("{:?}", id)));
```

Comment:
this line is untested according to tarpaulin

Reply:
Done
```
|| Uncovered Lines:
|| Tested/Total Lines:
|| src/primitives_identifiable.rs: 2/2
|| src/vec.rs: 123/123
|| src/vec_of.rs: 15/15
||
100.00% coverage, 140/140 lines covered
```